tests/cases/compiler/typePredicatesOptionalChaining2.ts

```diff
@@ -0,0 +1,15 @@
+// @strict: true
+
+type Person = { name: string; }
+
+const getName1 = (person?: Person): string => {
+  return typeof person?.name === 'string' ? person?.name : '';
+};
+
+const isString = (value: any): value is string => {
+  return typeof value === 'string';
+};
+
+const getName2 = (person?: Person): string => {
+  return isString(person?.name) ? person?.name : '';
```

Comment:
Shouldn't this be changed to this?
```suggestion
  return isString(person?.name) ? person.name : '';
```

Reply:
Both should be fine. If I recall correctly, optional chaining does not alter the type of the expression it is applied to.

Reply:
Ye, right - i thought that previously the error was also raised within the truthy branch of the conditional expression but it seems that this part was already working ok.